jwst/assign_wcs/assign_wcs.py

```diff
@@ -62,6 +63,13 @@ def load_wcs(input_model, reference_files={}, nrs_slit_y_range=None):
         wcs = WCS(pipeline)
         output_model.meta.wcs = wcs
         output_model.meta.cal_step.assign_wcs = 'COMPLETE'
+        if hasattr(output_model.meta, 'source_catalog'):
+            try:
+                output_model.meta.source_catalog.filename = os.path.basename(
+                    output_model.meta.source_catalog.filename)
+            except AttributeError:
+                pass
+
```

Comment:
This should not be in `assign_wcs.py`.  Setting a source catalog has nothing to do with `assign_wcs`.  I would completely remove this code.  Handle it where it is populated or actually used.  In addition to the two locations I mentioned aboe, it is also read here:

https://github.com/spacetelescope/jwst/blob/d086b9ac4120470e6b1b3c94dd6d786c47e67763/jwst/assign_wcs/util.py#L494

Reply:
Finally, if we want to catch the case were users put in bad values by hand into a FITS file, then I think we've lost the battle.  I'm happy to hard error there.  Users should be using a command line argument when running the pipeline if they have a user-supplied source catalog.